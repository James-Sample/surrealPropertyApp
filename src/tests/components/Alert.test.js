import React from "react";
import { render } from "@testing-library/react";
import Alert from "../../components/Alert";

describe("alert system", () => {
  xit("displays an error alert", () => {
    const { getByText } = render(<Alert message="Error!" />);
    expect(getByText(/Error/).textContent).toBe("Error!");
  });

  xit("displays a success message", () => {
    const { getByText } = render(<Alert message="Success!" success />);
    expect(getByText(/Success/).textContent).toBe("Success!");
  });

  xit("doesnt render an error or a success message if message props is empty", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
  });

  xit("diplays an error message if the message prop is Error", () => {
    const { getByText, asFragment } = render(<Alert message="Error!" />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Error/).textContent).toBe("Error!");
  });

  xit("diplays an error message if the message prop is success", () => {
    const { getByText, asFragment } = render(
      <Alert message="Success!" success />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Success/).textContent).toBe("Success!");
  });
});
