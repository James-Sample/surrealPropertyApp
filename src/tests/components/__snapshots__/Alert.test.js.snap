// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert system diplays an error message if the message prop is Error 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`alert system diplays an error message if the message prop is success 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Success!
  </div>
</DocumentFragment>
`;

exports[`alert system doesnt render an error or a success message if message props is empty 1`] = `<DocumentFragment />`;
